File: Numbers/change.rs
use std::env;

struct Change {
    dollars: u32,
    cents: u32
}

impl Change {
    fn take_dollars(&mut self, dollars: u32) -> u32 {
        let amount = self.dollars / dollars;
        self.dollars -= amount * dollars;
        amount
    }

    fn take_cents(&mut self, cents: u32) -> u32 {
        let amount = self.cents / cents;
        self.cents -= amount * cents;
        amount
    }
}

fn main() {
    let mut args = env::args();
    let cost = args.nth(1).expect("no cost provided");

    let mut change = match cost.find(".") {
        Some(dot) => {
            Change {
                dollars: cost[..dot].parse::<u32>().expect("ill formatted dollars"),
                cents: format!("{:0<2}", &cost[dot+1..])[..2] .parse::<u32>().expect("ill formatted cents")
            }
        },
        None => {
            Change {
                dollars: cost.parse::<u32>().expect("ill formatted dollars"),
                cents: 0
            }
        }
    };

    println!("hundreds {}", change.take_dollars(100));
    println!("fifties {}",  change.take_dollars(50));
    println!("twenties {}", change.take_dollars(20));
    println!("tens {}",     change.take_dollars(10));
    println!("fives {}",    change.take_dollars(5));
    println!("toonies {}",  change.take_dollars(2));
    println!("loonies {}",  change.take_dollars(1));
    println!("quarters {}", change.take_cents(25));
    println!("dimes {}",    change.take_cents(10));
    // i live in canada!
    if change.cents % 5 > 2 {
        change.cents += 3;
    }
    println!("nickles {}",  change.take_cents(5));
}

#[test]
fn test() {
}

File: Numbers/change.rb
dollars = ARGV[0].split('.')[0].to_i
cents = ARGV[0].split('.')[1].to_i

hundreds = dollars / 100
dollars -= hundreds * 100

fifties = dollars / 50
dollars -= fifties * 50

twenties = dollars / 20
dollars -= twenties * 20

tens = dollars / 10
dollars -= tens * 10

fives = dollars / 5
dollars -= fives * 5

twoonies = dollars / 2
dollars -= twoonies * 2

loonies = dollars

quarters = cents / 25
cents -= quarters * 25

dimes = cents / 10
cents -= dimes * 10

nickles = cents / 5
cents -= nickles * 5

pennies = cents

puts "100 x " + hundreds.to_s
puts "50 x " + fifties.to_s
puts "20 x " + twenties.to_s
puts "10 x " + tens.to_s
puts "5 x " + fives.to_s
puts "2 x " + twoonies.to_s
puts "1 x " + loonies.to_s
puts "0.25 x " + quarters.to_s
puts "0.10 x " + dimes.to_s
puts "0.05 x " + nickles.to_s
puts "0.01 x " + pennies.to_s
